force-app/main/default/classes/RecurrenceRule.cls

```diff
@@ -98,29 +99,19 @@ public with sharing class RecurrenceRule {
         }
 
         Set<Integer> daysToAdd = new Set<Integer>();
+        daysToAdd.addAll(new Set<Integer>{ 28, 29, 30, 31 });
```

Comment:
I don't think we can just add all, we still need the when 29 but only when not last day of month. For example test repeating on dec 31st, should be jan 31, feb 28, march 31 etc.

Reply:
@jjbennett, when we add months salesforce takes care of checking the number of days in a month. I can attach a screen shot of what I am seeing in the UI. So I made session start date as 12/31/20 and let it recur for 5 months. Since Feb 2021 has 28 days the date shows up as 28. Let me know if you want to talk through this.

Reply:
@jjbennett , I had to add back the logic where the byMonthDays will always return 28,29,30,31 but will only be used when it is last day of the month. Now when we repeat on Dec 31st the values are what you had mentioned in the comment above. I had to remove a couple of tests from recurrence rule since they tested bymonthrule based on the date and it was no longer valid. I also went ahead and added a new test to the Monthly_Iterator class. I hope this solution has solved all the issues we had